velox/functions/lib/benchmarks/LikeFunctionsBenchmark.cpp

```diff
@@ -0,0 +1,262 @@
+        if (!input->isNullAt(j)) {
+          auto inputAtIdx = input->valueAt(j);
+          like(inputAtIdx.str(), patternString);
+        }
+      }
+    }
+  }
+
+  void run(vector_size_t vectorSize, int numRuns, PatternKind patternKind) {
+    auto like = [&](std::optional<std::string> str,
+                    std::optional<std::string> pattern) {
+      return evaluateOnce<bool>("like(c0, '" + *pattern + "')", str);
+    };
+
+    folly::BenchmarkSuspender kSuspender;
+    FunctionBenchmarkBase benchmarkBase;
+    kSuspender.dismiss();
+    std::vector<std::string> inputVector;
+    inputVector.reserve(vectorSize);
+    std::vector<std::string> patternVector;
+    patternVector.reserve(vectorSize);
+    for (vector_size_t i = 0; i < vectorSize; i++) {
+      auto fixedPatternString = generateRandomString(alphaNumCharSet_);
+      inputVector.emplace_back(generateInput(
+          generateRandomString(alphaNumCharSet_),
+          fixedPatternString,
+          patternKind));
+      patternVector.emplace_back(
+          generatePattern(patternKind, fixedPatternString));
+    }
+    auto pattern = benchmarkBase.maker().flatVector(patternVector);
+    auto input = benchmarkBase.maker().flatVector(patternVector);
+
+    for (int i = 0; i < numRuns; i++) {
+      for (vector_size_t j = 0; j < input->size(); j++) {
+        auto inputAtIdx = input->valueAt(j);
+        auto patternAtIdx = pattern->valueAt(j);
+        like(inputAtIdx, patternAtIdx);
+      }
+    }
+  }
+
+  void TestBody() {
+    return;
+  }
+
+ protected:
+  const char* alphaNumCharSet_ =
+      "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  const char* wildcardCharSet_ = "%_";
+  const char* anyWildcardCharSet_ = "%";
+  const char* singleWildcardCharSet_ = "_";
+};
+
+} // namespace facebook::velox::functions::test
+
+std::unique_ptr<functions::test::LikeFunctionsBenchmark> benchmark;
+
+BENCHMARK(wildcardPatternOne) {
+  vector_size_t vectorSize = 100;
+  PatternKind patternKind = PatternKind::kWildcardOne;
+  benchmark->run(vectorSize, FLAGS_num_runs, patternKind);
+}
+
+BENCHMARK(wildcardPatternTwo) {
+  vector_size_t vectorSize = 100;
+  PatternKind patternKind = PatternKind::kWildcardTwo;
+  benchmark->run(vectorSize, FLAGS_num_runs, patternKind);
+}
+
+BENCHMARK(fixedPattern) {
+  vector_size_t vectorSize = 100;
+  PatternKind patternKind = PatternKind::kFixed;
+  benchmark->run(vectorSize, FLAGS_num_runs, patternKind);
+}
+
+BENCHMARK(prefixPattern) {
+  vector_size_t vectorSize = 100;
+  PatternKind patternKind = PatternKind::kPrefix;
+  benchmark->run(vectorSize, FLAGS_num_runs, patternKind);
+}
+
+BENCHMARK(suffixPattern) {
+  vector_size_t vectorSize = 100;
+  PatternKind patternKind = PatternKind::kSuffix;
+  benchmark->run(vectorSize, FLAGS_num_runs, patternKind);
+}
+
+BENCHMARK(tpchQuery2) {
+  auto partTypeVector =
+      genTpchPart(FLAGS_num_rows)->childAt(4)->asFlatVector<StringView>();
```

Comment:
I think we need to hold `genTpchPart(FLAGS_num_rows)` in a variable. Same below.

Reply:
Thanks for the suggestion, changed accordingly.